### PR TITLE
Direct field access for RouteContext#env

### PR DESCRIPTION
**Pull request by ByteAlex**
Hi,

since data just got updated to use direct field access instead, but env is facinig the same limitiations, I thought a PR to fix that would be a good idea.

### CODE REVIEW

**Comment by nilslice** [comment]
I'm not opposed to this, but would like to know what @zebp thinks. I can see where it would make some things easier and I like that it's consistent with the `data` change. Just want to double-check that there aren't any edge cases where this could cause issue. 

It could also make #76 less of an issue. I don't recall where we left off, but I think there was discussion of adding `Clone` to `Env` if it isn't already.

File: worker/src/router.rs [hunk 1]
```diff
@@ -66,6 +66,7 @@ impl<D> RouteContext<D> {
 
     /// Get the `Env` for this Worker. Typically users should opt for the `secret`, `var`, `kv` and
     /// `durable_object` methods on the `RouteContext` instead.
+    #[deprecated(since = "0.0.8", note="please use the `env` field directly")]
```
Comment (zebp) [inline worker/src/router.rs:69]: The version should probably be `0.0.9` as this correct for `0.0.8` and the formatting for the `note` argument isn't correct.

**Review by zebp — APPROVED** [review]
I think this looks good, I can't think of any situation off the top of my head where this change would break.

File: worker/src/router.rs (hunk 1 above)
Comment (nilslice) [reply worker/src/router.rs:69]: We can probably keep it as `0.0.8`, unless you specifically want to wait to include this in a future release (not the upcoming `0.0.8`)?

Comment (zebp) [reply worker/src/router.rs:69]: Woops! I misremembered what version we were on. Never mind the version then.

**Review by nilslice — APPROVED** [review]
LGTM, thank you! I couldn't find any example code that references this which would need updating.. but happy to have someone take a second look.

**Comment by ByteAlex** [comment]
Hi, sorry I just looked into this again - I copied the formatting from the above "data" field, which apparently has wrong formatting too. Eventually you want to fix it there too?

**Comment by zebp** [comment]
This PR looks good for merge, the CI failure seems irrelevant and a problem on our end. I also couldn't find any out of date example code.

> Hi, sorry I just looked into this again - I copied the formatting from the above "data" field, which apparently has wrong formatting too. Eventually you want to fix it there too?

No that's fine, it's already fixed on `main`.

**Pull request merged**